src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3TestBase.cs

```diff
@@ -451,36 +455,48 @@ public async ValueTask<Http3ControlStream> CreateControlStream(int? id)
 
         internal ValueTask<Http3RequestStream> CreateRequestStream()
         {
-            var stream = new Http3RequestStream(this, Connection);
+            if (!_streamContextPool.TryDequeue(out var testStreamContext))
+            {
+                testStreamContext = new TestStreamContext(canRead: true, canWrite: true, this);
+            }
+            testStreamContext.Initialize(GetStreamId(0x00));
```

Comment:
`TestStreamContext` is now reused to simulate how QUIC stream contexts are reused.

It is a bit of extra complexity to our tests, but I want to be able to test pooling Http3Streams without having to use Quic transport.